Hazel/src/Hazel/Debug/Instrumentor.h

```diff
@@ -105,6 +100,16 @@ namespace Hazel {
 
 	private:
 
+		Instrumentor()
+			: m_CurrentSession(nullptr)
+		{
+		}
+
+		~Instrumentor()
+		{
+			InternalEndSession();
```

Comment:
Shouldn't it call `EndSession()`? For now it's fine, but I'm thinking about multi threaded cases...

Reply:
Yeah good catch. I missed the comment above `InternalEndSession()` that notes you must already own the lock on mutex before calling it. Is there a reason why it doesn't own the lock itself? It would make sense for the lock to be created where it's needed rather than expecting it to be/hoping it is. If it was moved to inside `InternalEndSession()` then it would require the lock in `BeginSession(...)` to move to after the check to see if `m_CurrentSession` is active or not to ensure the rest of the method owns a lock.

Reply:
Fair enough, that makes sense. I've pushed a new commit that calls `EndSession()` instead of `InternalEndSession()` from the destructor just so the fix is complete.